### ISSUE TITLE
Podman container support

### ISSUE THREAD

**Issue by q2dg** [GNS3/gns3-server#1811]
Hello. Have you considered Podman container support? Its API is very similar to Docker's one. 
I would be great for running Gns3 in RedHat/Fedora/CentOS hosts, where Docker isn't officially available. 
Furthermore, it increases the security of all. 
Thanks!

**Comment by elico** [comment]
@grossmj @q2dg any progress with this?
I wouldn't worry about security with Docker, the whole rootless point is good but up to the point when you really trust the product and the code.
Docker is pretty mature to be trusted enough for GNS3.
The main issue is compatibility on:
RHEL 8+
Fedora 33+
CentOS 8+
Oracle Enterprise Linux 8+
....

Which I use all the above.

**Comment by q2dg** [comment]
Yes, after more than two years without any official response, this compatibility issue starts to be a blocking one,

**Comment by rb83** [comment]
It almost works.

But at the moment podman will always create network interfaces in the container, even when requested not to do so. This prevents ubridge from moving interfaces to the container's namespace & renaming them, since they already exist with the same name inside the container. This is a podman issue, not a GNS3 issue.

With regards to security, it's completely independent of wether one uses Docker or Podman, as GNS3 currently requires the containers to be rootfull anyway (also see #1944).

**Comment by benyaminl** [comment]
does this include podman.socket needed?

**Comment by rb83** [comment]
@benyaminl that error message is misleading (the GNS3 GUI thinks the container's stdout error message is related, but it's not). The process fails earlier, as described in my previous message.

You have to use docker for the time being.

**Comment by benyaminl** [comment]
@rb83 So sad :') 

Thanks for clear things up.. Probably I will use full Qemu VM for now. Thank you!

**Comment by q2dg** [comment]
Well, it seems Podman's behaviour, which was a no-go in this issue until resolved, is finally resolved: https://github.com/containers/podman/issues/16716

**Comment by benyaminl** [comment]
@q2dg have you test it on newest GNS3? Does it work?

**Comment by q2dg** [comment]
Not yet. I'll update this thread with new information when I get it

**Comment by grossmj** [comment]
[like]  Jeremy Grossmann reacted to your message:
________________________________

**Comment by RushingAlien** [comment]
Still same /etc/hosts issue looks like

**Comment by q2dg** [comment]
I'm waiting to new Podman's v.5.0, where there are some changes in its internal network stack. When released I'll give feedback

**Comment by RushingAlien** [comment]
![gambar](https://github.com/user-attachments/assets/f318d946-0a28-427f-af4a-ca0f211a2609)

still getting this issue. Though i think the fix is simply adding --privileged to the command

**Comment by RushingAlien** [comment]
Hey! So through the closed ticket, it seems to me we need to run the container with --no-hosts?

**Comment by RushingAlien** [comment]
Hey! So through the closed ticket, it seems to me we need to run the container with --no-hosts?

**Comment by yueguobin** [comment]
Since 'docker-ce-rootless-extras' is available, I think it's a viable solution. All we need to do is install the package and update '/var/run/docker.sock' to '/run/user/$ID/docker.sock'.
